design/open/high-level-python-wrapper.md

```diff
@@ -107,15 +109,17 @@ class TagManager:
     def list(max_amount: Optional[int], after: str ='', prefix: str ='') -> Generator[Tag]: ...
 
 
-class Object:
+class StoredObject:
     def __init__(self, repository_id: str, reference_id: str, path: str, client: Client = DefaultClient): ...
-    def open(self, pre_signed: Optional[bool] = None) -> TextIO | BinaryIO: ...
+    def open(self, mode: Literal['r', 'rb'] = 'r', pre_signed: Optional[bool] = None) -> TextIO | BinaryIO: ...
     def stat(self) -> ObjectInfo: ...
 
 
-class WritableObject(Object):
-    def create(self, data: bytes | str | TextIO | BinaryIO, path: str, pre_signed: Optional[bool] = None) -> ObjectInfo: ...
-    def delete(self, object_paths: str | Iterable[str]): ...
+class WritableObject(StoredObject):
+    def create(self, data: bytes | str | TextIO | BinaryIO, path: str, pre_signed: Optional[bool] = None,
+        content_type: Optional[str] = None, metadata: Optional[dict[str, str]] = None, 
```

Comment:
If I specify `ContentType` in `metadata`, which takes precedence?  (Probably the same as the lakeFS API defaults)